### PR TITLE
Rename settingsBase, -Vey, -Github to -BaseType, -Veyor, -GitHub

### LINKED ISSUE #180: Rename some of the fields in the `Settings` data type
This data type:

https://github.com/kowainik/summoner/blob/0a533d01bf1800a4fbe1323f40efdb66a8bbae0b/src/Summoner/Settings.hs#L16-L42

Historically some names are not perfect. Let's do the following:

1. `settingsGithub` to `settingsGitHub`
2. `settingsAppVey` to `settingsAppVeyor`
3. `settingsBase` to `settingsBaseType`

Should be ok for now :+1:

### PR DESCRIPTION
**Pull request by Bargsteen**
Resolves #180 

I hope that I filled out the checklist correctly. Otherwise, let me know :)
## Checklist:

- [ ] I've updated the [CHANGELOG](https://github.com/kowainik/summoner/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [x] All new and existing tests pass.
- [x] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [x] I've used the [`stylish-haskell` file](https://github.com/kowainik/summoner/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.
- [ ] I changed code related to the generated project.
  - [ ] I created a new project using `summoner` and I checked that this specific feature is working as expected.

### CODE REVIEW

File: src/Summoner/Template/Cabal.hs [hunk 1]
```diff
@@ -79,8 +79,8 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
           exposed-modules:     $libModuleName
                                $preludeMod
 
-          build-depends:       $settingsBase
-                             $commaPreludeLibrary
+          build-depends:         $settingsBaseType
```
Comment (vrom911) [inline src/Summoner/Template/Cabal.hs]: Oops, I guess space is accident
```suggestion
          build-depends:       $settingsBaseType
```

File: src/Summoner/Template/Cabal.hs [hunk 2]
```diff
@@ -79,8 +79,8 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
...
+                               $commaPreludeLibrary
```
Comment (vrom911) [inline src/Summoner/Template/Cabal.hs]:
```suggestion
                             $commaPreludeLibrary
```

Comment (Bargsteen) [reply src/Summoner/Template/Cabal.hs]: It is now fixed. I rebased to keep the commit history clean.

**Comment by Bargsteen** [comment]
It is now fixed. I rebased to keep the commit history clean.

File: src/Summoner/Template/Cabal.hs [hunk 3]
```diff
@@ -79,8 +79,8 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
           exposed-modules:     $libModuleName
                                $preludeMod
 
-          build-depends:       $settingsBase
-                             $commaPreludeLibrary
+          build-depends:       $settingsBaseType
+                               $commaPreludeLibrary
```
Comment (vrom911) [inline src/Summoner/Template/Cabal.hs]: This two spaces shouldn't be added. Basically, this line contains the comma in it, so after rendering it looks like this:
```
          build-depends:       base
                             , relude
```

But, if you add this two spaces you will get:
```
          build-depends:       base
                               , relude
```

which is not correct.

File: src/Summoner/Template/Cabal.hs [hunk 4]
```diff
@@ -97,9 +97,9 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
           hs-source-dirs:      app
           main-is:             Main.hs
 
-          build-depends:       $settingsBase
-                             $commaRepo
-                             $commaPreludeLibrary
+          build-depends:       $settingsBaseType
+                               $commaRepo
+                               $commaPreludeLibrary
```
Comment (vrom911) [inline src/Summoner/Template/Cabal.hs]: The same here. These are line with commas in it. So, no need it two extra spaces.

File: src/Summoner/Template/Cabal.hs [hunk 5]
```diff
@@ -120,9 +120,9 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
           hs-source-dirs:      test
           main-is:             Spec.hs
 
-          build-depends:       $settingsBase
-                             $commaRepo
-                             $commaPreludeLibrary
+          build-depends:       $settingsBaseType
+                               $commaRepo
+                               $commaPreludeLibrary
```
Comment (vrom911) [inline src/Summoner/Template/Cabal.hs]: And here

File: src/Summoner/Template/Cabal.hs [hunk 6]
```diff
@@ -143,10 +143,10 @@ cabalFile Settings{..} = File (toString settingsRepo ++ ".cabal") cabalFileConte
           hs-source-dirs:      benchmark
           main-is:             Main.hs
 
-          build-depends:       $settingsBase
+          build-depends:       $settingsBaseType
                              , gauge
-                             $commaRepo
-                             $commaPreludeLibrary
+                               $commaRepo
+                               $commaPreludeLibrary
```
Comment (vrom911) [inline src/Summoner/Template/Cabal.hs]: And here.

**Comment by Bargsteen** [comment]
Ah okay, I get it. *Now* it should be fixed then. :)

**Pull request merged**